command/ca/acme/eab/list.go

```diff
@@ -0,0 +1,178 @@
+package eab
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"os/signal"
+	"syscall"
+
+	"github.com/pkg/errors"
+	"github.com/smallstep/certificates/ca"
+	"github.com/smallstep/cli/flags"
+	"github.com/smallstep/cli/utils/cautils"
+	"github.com/urfave/cli"
+	"go.step.sm/cli-utils/errs"
+)
+
+func listCommand() cli.Command {
+	return cli.Command{
+		Name:   "list",
+		Action: cli.ActionFunc(listAction),
+		Usage:  "list all ACME External Account Binding Keys",
+		UsageText: `**step beta ca acme eab list** <provisioner> [<reference>]
+[**--limit**=<number>] [**--admin-cert**=<file>] [**--admin-key**=<file>]
+[**--admin-provisioner**=<string>] [**--admin-subject**=<string>]
+[**--password-file**=<file>] [**--ca-url**=<uri>] [**--root**=<file>]
+[**--context**=<name>]`,
+		Flags: []cli.Flag{
+			flags.Limit,
+			flags.NoPager,
+			flags.AdminCert,
+			flags.AdminKey,
+			flags.AdminProvisioner,
+			flags.AdminSubject,
+			flags.PasswordFile,
+			flags.CaURL,
+			flags.Root,
+			flags.Context,
+		},
+		Description: `**step beta ca acme eab list** lists all ACME External Account Binding (EAB) Keys.
+
+Output will go to stdout by default. If many EAB keys are stored in the ACME provisioner, output will be sent to $PAGER (when set). 
+
+## POSITIONAL ARGUMENTS
+
+<provisioner>
+: Name of the provisioner to list ACME EAB keys for
+
+<reference>
+: (Optional) reference (from external system) for the key to be listed
+
+
+## EXAMPLES
+
+List all ACME External Account Binding Keys:
+'''
+$ step beta ca acme eab list my_acme_provisioner
+'''
+
+Show ACME External Account Binding Key with specific reference:
+'''
+$ step beta ca acme eab list my_acme_provisioner my_reference
+'''
+`,
+	}
+}
+
+func listAction(ctx *cli.Context) (err error) {
+	if err := errs.MinMaxNumberOfArguments(ctx, 1, 2); err != nil {
+		return err
+	}
+
+	args := ctx.Args()
+	provisioner := args.Get(0)
+
+	reference := ""
+	if ctx.NArg() == 2 {
+		reference = args.Get(1)
+	}
+
+	client, err := cautils.NewAdminClient(ctx)
+	if err != nil {
+		return errors.Wrap(err, "error creating admin client")
+	}
+
+	var out io.WriteCloser
+	var cmd *exec.Cmd
+
+	usePager := true
+	if ctx.IsSet("no-pager") {
+		usePager = !ctx.Bool("no-pager")
+	}
+
+	// the pipeSignalHandler goroutine ensures that the parent process is closed
+	// whenever one of its childs is killed.
```

Comment:
children are killed

Reply:
😅 Fixed in 686e78f8613be934418156b4c357eacb32e06f80.